client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/backend/initialize/FeatureFlagsDto.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SonarLint Core - Client API
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.clientapi.backend.initialize;
+
+/**
+ * Optional features toggles of the backend. To ease transition or to accommodate different client needs.
+ */
+public class FeatureFlagsDto {
+
+  private final boolean shouldManageSmartNotifications;
+  @Deprecated
+  // not controllable anymore, it is the backend's responsibility to decide
+  private final boolean taintVulnerabilitiesEnabled;
```

Comment:
Instead of moving this, I think we could remove it

Reply:
OK, I will merge as is, and we can remove this at the next opportunity.